internal/provider/kubernetes/routes.go

```diff
@@ -177,16 +177,16 @@ func (r *gatewayAPIReconciler) processGRPCRoutes(ctx context.Context, gatewayNam
 					authFilter, ok := resourceMap.authenFilters[key]
```

Comment:
this logic doesnt look right, shouldnt it look more like https://github.com/envoyproxy/gateway/blob/a731995bb3b56366a98eb0b256cc3bec3726e46f/internal/provider/kubernetes/routes.go#L355 which is based on `switch`.

tldr - its not an actual error if the filter is not a AuthenticationFilter and we are trying to fetch something that doesnt exist

Reply:
fixed